src/Compilers/Core/Portable/Compilation/IOperation.cs

```diff
@@ -35,95 +44,164 @@ public interface IOperation
-        ConditionalChoiceExpression,
-        NullCoalescingExpression,
-        LambdaExpression,
-        ObjectCreationExpression,
-        TypeParameterObjectCreationExpression,
-        ArrayCreationExpression,
-        DefaultValueExpression,
-        InstanceReferenceExpression,
-        BaseClassInstanceReferenceExpression,
-        ClassInstanceReferenceExpression,
-        IsExpression,
-        TypeOperationExpression,
-        AwaitExpression,
-        AddressOfExpression,
-        AssignmentExpression,
-        CompoundAssignmentExpression,
-        ParenthesizedExpression,
-
-        UnboundLambdaExpression,
-        EventAssignmentExpression,
+        None = 0x0,
+
+        /// <summary>Indicates an <see cref="IInvalidStatement"/>.</summary>
+        InvalidStatement = 0x1,
+        /// <summary>Indicates an <see cref="IBlockStatement"/>.</summary>
+        BlockStatement = 0x2,
+        /// <summary>Indicates an <see cref="IVariableDeclarationStatement"/>.</summary>
+        VariableDeclarationStatement = 0x3,
+        /// <summary>Indicates an <see cref="ISwitchStatement"/>.</summary>
+        SwitchStatement = 0x4,
+        /// <summary>Indicates an <see cref="IIfStatement"/>.</summary>
+        IfStatement = 0x5,
+        /// <summary>Indicates an <see cref="ILoopStatement"/>.</summary>
+        LoopStatement = 0x6,
+        /// <summary>Indicates an <see cref="IReturnStatement"/>.</summary>
+        YieldBreakStatement = 0x9,
+        /// <summary>Indicates an <see cref="ILabelStatement"/>.</summary>
+        LabelStatement = 0xa,
+        /// <summary>Indicates an <see cref="IBranchStatement"/>.</summary>
+        BranchStatement = 0xc,
+        /// <summary>Indicates an <see cref="IEmptyStatement"/>.</summary>
+        EmptyStatement = 0xd,
+        /// <summary>Indicates an <see cref="IThrowStatement"/>.</summary>
+        ThrowStatement = 0xe,
+        /// <summary>Indicates an <see cref="IReturnStatement"/>.</summary>
+        ReturnStatement = 0xf,
+        /// <summary>Indicates an <see cref="ILockStatement"/>.</summary>
+        LockStatement = 0x10,
+        /// <summary>Indicates an <see cref="ITryStatement"/>.</summary>
+        TryStatement = 0x11,
+        /// <summary>Indicates an <see cref="ICatchClause"/>.</summary>
+        CatchClause = 0x12,
+        /// <summary>Indicates an <see cref="IUsingWithDeclarationStatement"/>.</summary>
+        UsingWithDeclarationStatement = 0x13,
+        /// <summary>Indicates an <see cref="IUsingWithExpressionStatement"/>.</summary>
+        UsingWithExpressionStatement = 0x14,
+        /// <summary>Indicates an <see cref="IReturnStatement"/>.</summary>
+        YieldReturnStatement = 0x15,
+        /// <summary>Indicates an <see cref="IFixedStatement"/>.</summary>
+        FixedStatement = 0x16,
+        // LocalFunctionStatement = 0x17,
+
+        /// <summary>Indicates an <see cref="IExpressionStatement"/>.</summary>
+        ExpressionStatement = 0x18,
```

Comment:
I probably missed the discussion during review, but I think it'd be nice to leave some room between different type of operations for possible future expansion (i.e. statement vs. expression vs. neither, etc.)